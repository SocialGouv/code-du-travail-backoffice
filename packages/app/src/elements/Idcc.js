import styled from "@emotion/styled";
import React from "react";

import { validateMandatoryNullableString } from "../props/validators";

const Container = styled.div`
  background-color: var(--color-label-background);
  border: solid 1px var(--color-border);
  border-radius: 0.25rem;
  cursor: help;
  font-size: 0.8rem;
  font-weight: 600;
  margin-right: 0.75rem;
  max-width: 3rem;
  min-width: 3rem;
  opacity: 0.75;
  padding: 0.25rem 0;
  position: relative;
  text-align: center;
  user-select: none;
  width: 3rem;
  white-space: nowrap;

  > div {
    background-color: var(--color-eerie-black);
    border-radius: 0.25rem;
    box-shadow: 0 0 0.5rem var(--color-eerie-black);
    color: white;
    display: none;
    font-size: 0.8rem;
    font-weight: 600;
    left: -1px;
    padding: 0.35rem 0.6rem 0.4rem;
    position: absolute;
    text-align: left;
    bottom: 2rem;
    width: 22rem;
    white-space: normal;
    z-index: 2;
  }

  :hover {
    opacity: 1;

    > div {
      display: block;
    }
  }
`;

const Idcc = ({ code, name, ...props }) => (
  <Container {...props}>
    {code === null ? "CDT" : code}
    <div data-testid="tooltip">{name === null ? "Code du travail" : name}</div>
  </Container>
);

Idcc.propTypes = {
  code: validateMandatoryNullableString,
  name: validateMandatoryNullableString,
};

export default Idcc;
