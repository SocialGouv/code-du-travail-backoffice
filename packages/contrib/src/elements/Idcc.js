import styled from "@emotion/styled";
import React from "react";
import ReactTooltip from "react-tooltip";

const Container = styled.div`
  background-color: var(--color-label-background);
  border: solid 1px var(--color-black-leather-jacket);
  border-radius: 1rem;
  cursor: help;
  font-size: 0.8rem;
  font-weight: 600;
  margin-right: 0.75rem;
  opacity: 0.75;
  padding: 0.25rem 0.5rem;
  user-select: none;
  white-space: nowrap;

  :hover {
    opacity: 1;
  }
`;

const Tooltip = styled(ReactTooltip)`
  max-width: 22rem;
  padding: 0.25rem 0.5rem 0.45rem;
  white-space: normal;
`;

const Idcc = ({ code, name, ...props }) => (
  <Container data-tip={code === undefined ? "Code du travail" : name} {...props}>
    {code === undefined ? "CDT" : `IDCC: ${code}`}
    <Tooltip />
  </Container>
);

export default Idcc;
