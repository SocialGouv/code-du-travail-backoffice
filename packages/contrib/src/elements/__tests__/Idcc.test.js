import React from "react";
import { render } from "react-testing-library";
import Idcc from "../Idcc";

// Ignore styled-wrapped ReactTooltip className prop warning
console.warn = jest.fn();

describe("[Contrib] elements/<Idcc />", () => {
  it("should match snapshot", () => {
    const data = { idcc: "1234", name: "A labor agreement title" };
    const { container } = render(<Idcc data={data} />);

    expect(container).toMatchSnapshot();
  });
});
