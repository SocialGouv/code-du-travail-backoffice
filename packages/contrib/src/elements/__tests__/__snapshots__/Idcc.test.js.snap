// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] elements/<Idcc /> should match snapshot 1`] = `
<div>
  <div
    class="Idcc__Container-owzimc-0 dbOONK"
    currentitem="false"
    data-tip="A labor agreement title"
  >
    1234
    <div
      class="__react_component_tooltip place-top type-dark "
      data-id="tooltip"
    />
  </div>
</div>
`;
